google/cloud/storage/client_sign_url_test.cc

```diff
@@ -107,6 +107,17 @@ TEST_F(CreateSignedUrlTest, V2SignRemote) {
   EXPECT_THAT(*actual, HasSubstr(expected_signed_blob_safe));
 }
 
+/// @test Verify that CreateV2SignedUrl() respects the custom endpoint.
+TEST_F(CreateSignedUrlTest, V2SignCustomEndpoint) {
+  std::string custom_endpoint = "https://storage.mydomain.com";
+
+  Client client(Options{}.set<RestEndpointOption>(custom_endpoint));
```

Comment:
The client library tries to use a remote service to create signed URLs if the client is not initialized with credentials that can signed them locally.  And many of our CI builds don't have permissions to create signed URLs.  You are better off with something like:

```suggestion
  Client client(Options{}
      .set<RestEndpointOption>(custom_endpoint)
      .set<UnifiedCredentialsOption>(
      MakeServiceAccountCredentials(kJsonKeyfileContents)));
 ```